### PR TITLE
fix(vite): remove `postcss-url` and duplicate `postcss-import`

### LINKED ISSUE #15765: Load a external font failed the server response is incorrect
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.12.1`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `pnpm@7.17.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction
[https://github.com/yyyanghj/nuxt-issue-9357](https://github.com/yyyanghj/nuxt-issue-9357)

### Describe the bug

Importing a CSS file to load a font failed. The server responds with an html file instead of a font file.

If I import the font.css in the `<script>`, everything is fine.

```vue
<script>
// uncomment next line works well.
// import "@fontsource/lora";
</script>

<style>
/* this does not work. */
@import "@fontsource/lora";

body {
  font-family: "Lora";
  font-size: 100px;
}
</style>

```

https://user-images.githubusercontent.com/34713301/204149097-89818bda-76e4-4d54-bf31-563142e19702.mov

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/15765

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This was a fun deep-dive.

Investigating the linked issue, I found that we are adding `postcss-import` - _and so is vite_. In addition, I believe the key features `postcss-url` (which was causing the bug) are in fact already covered by Vite's list of features:

> Vite improves @import resolving for Sass and Less so that Vite aliases are also respected. In addition, relative url() references inside imported Sass/Less files that are in different directories from the root file are also automatically rebased to ensure correctness.
> https://vitejs.dev/guide/features.html#css-pre-processors

Accordingly, this PR removes both plugins (or rather, doesn't add them).

I suspect removing these two plugins will improve our performance as well as fix the bug. I would also consider we reconsider whether to include `cssnano` or defer to vite's esbuild/lightningcss minificiation step in future (cc: antfu).

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/webpack/src/utils/postcss.ts [hunk 1]
```diff
@@ -25,7 +26,7 @@ const orderPresets = {
   }
 }
 
-export const getPostcssConfig = (nuxt: Nuxt) => {
+export const getPostcssConfig = async async async async (nuxt: Nuxt) => {
```
Comment (johannschopplich) [inline packages/webpack/src/utils/postcss.ts:29]: This seems to be a typo? 🙂 
```suggestion
export const getPostcssConfig = async (nuxt: Nuxt) => {
```

Comment (danielroe) [reply packages/webpack/src/utils/postcss.ts]: thank you! ❤️

File: packages/schema/src/config/postcss.ts [hunk 2]
```diff
@@ -11,29 +9,6 @@ export default defineUntypedSchema({
...
-            })
-          })
-          : val
-      },
-
-      /**
-       * https://github.com/postcss/postcss-url
-       */
-      'postcss-url': {},
-
```
Comment (HigherOrderLogic) [inline packages/schema/src/config/postcss.ts:36]: Shouldnt we leave the type as is and mark it as deprecated to avoid breaking changes?

**Comment by mrleblanc101** [comment]
Shouldn't this mention/link to #20533 since all the related issue were merged into that one ?

**Review by antfu — APPROVED** [review]

**Comment by antfu** [comment]
For CSS minification, I am not sure if users would want full control over which tool to use, maybe could be an option as `'cssnano' | 'esbuild' | 'lightningcss'` where we built a thin layer of adapter?

**Comment by danielroe** [comment]
I think that would be a very nice idea.

**Comment by nathanchase** [comment]
@danielroe @antfu 

> For CSS minification, I am not sure if users would want full control over which tool to use, maybe could be an option as `'cssnano' | 'esbuild' | 'lightningcss'` where we built a thin layer of adapter?

Full control is good, but also with the most performant (sensible) default (likely `lightningcss`). I was recently looking at whether or not something like [nuxt-modules/critters](https://github.com/nuxt-modules/critters) or [nuxt-beastcss](https://github.com/bnznamco/nuxt-beastcss) was even necessary, or if Nuxt 3 already accomplishes the same goals. 

There's not much in the docs other than Nuxt is pre-configured to use `cssnano` for "minification and purge" (at the end of "Using Postcss") https://nuxt.com/docs/getting-started/styling#using-postcss - but nothing specifically about best practices of inlining critical css. 

I know there's the `inlineSSRStyles` [experimental flag](https://nuxt.com/docs/api/nuxt-config#inlinessrstyles) defaulting to true, but unsure if that's doing the same thing that critters / beastcss are or not.

Anyway, ideally Nuxt 3 takes care of all of the above for us (inlined critical CSS, fastest minification and purging), and maybe explains the optimal pre-configured choice (lightning) - but then lets us opt for other less performant options if it's somehow a project requirement.

**Pull request merged**